### PR TITLE
Tutorial animations

### LINKED ISSUE #135: Tutorial: Animations.jl
We need another tutorial or some kind of documentation for #125 once that is finished.

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [x] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
  - not applicable
- [x] Did I update the `TODO.md` (if applicable)?
  - not applicable
- [ ] Did I make sure to only change the part of the file where I introduced a new change/feature?
  - I have also made some documentation changes that I saw
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
  - not applicable
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
  - not applicable
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
  - not applicable
- [x] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #135 

**How did you address these issues with this PR? What methods did you use?**
This adds a tutorial no. 6 as #126 will be no. 5 
In that tutorial I explain how to create this loading animation.
loading

### CODE REVIEW

File: docs/src/tutorials/tutorial_6.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+# **Tutorial 6:** Using Animations.jl to Create something with more Pep!
+
+In the last couple of tutorials you've learned the basics of Javis and some of the more advanced stuff like [`SubAction`](@ref).
+
+## Our goal
+
+The goal of this tutorial is to explain a new feature we have added in v0.2 of Javis. Before this every animation as basically linear.
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:7]: "every animation as basically linear." -> "was basically linear"

File: docs/src/tutorials/tutorial_6.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+What I mean by this is: If you move an object from `A` to `B` using [`Translation`](@ref) it would do so in a linear and therefore boring fashion.
+
+We'll create an interesting loading animation for this. It consists of five circles which move from the center to the outside rotate around the center and back to the origin. During this process they appear and disappear as well as changing color.
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:10]: Maybe just say "In this tutorial, we'll create a loading animation that looks like this:" and show the final gif below it.

File: docs/src/tutorials/tutorial_6.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+## Learning Outcomes
+
+This tutorial shows some more power of subactions that we introduced in v0.2. Combined with the power of the awesome library [Animations.jl](https://github.com/jkrumbiegel/Animations.jl) you have very fine grained control over the objects you animate and can benefit from the easiness of easing functions. 😄
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:14]: "and can benefit from the easiness of easing functions." -> "and the ease of its easing functions."

File: docs/src/tutorials/tutorial_6.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+Today you'll learn how to
+- use easing functions to have animations with pep
+- have full control over the animation of objects
+- create a beautiful (if I can say so myself) loading animation
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:19]: :heart:

File: docs/src/tutorials/tutorial_6.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    video,
+    [BackgroundAction(1:200, ground), actions...],
+    pathname = "loading.gif",
+)
+```
+
+![Linear movement](assets/loading_circle_linear.gif)
+
+I would say that this looks a bit dull. Let us rotate it with varying speeds. For this I'll use the `sineio` easing function. You can try another easing function.
+They are all described [here](https://jkrumbiegel.github.io/Animations.jl/stable/#Easings-1).
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:56]: Maybe add a small note about what easing functions are?

File: docs/src/tutorials/tutorial_6.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+First of all we need `Animations` for this and I also load `Colors` here for later.
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:58]: You should mention needing these two additional packages in the introduction like how I did in Tutorial 5 on scaling.

File: docs/src/tutorials/tutorial_6.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+javis(
+    video,
+    [BackgroundAction(1:200, ground), actions...],
+    pathname = "loading.gif",
+)
+```
+
+![Rotation with varying speed](assets/loading_circle_sineio.gif)
+
+I think that looks more interesting. The [Animations.jl](https://jkrumbiegel.github.io/Animations.jl/stable) package is very powerful so you might wanna head over to their documentation and try different things.
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:94]: I like that you explain Animations a little here but I think you should also go further into detail about what Animations and Colors are doing.

File: docs/src/tutorials/tutorial_6.md [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+
+One translating from the origin to the right 
+
+```julia
+translate_anim = Animation(
+    [0, 1], # must go from 0 to 1
+    [O, Point(150, 0)],
+    [sineio()],
+)
+```
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:112]: Maybe mention what the `sineio()` easing function is doing exactly?

File: docs/src/tutorials/tutorial_6.md [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+            SubAction(1:30, translate_anim, translate()),
+            SubAction(31:120, rotate_anim, rotate_around(Point(-150, 0))),
+            SubAction(121:150, translate_back_anim, translate()),
+        ],
+    ) for frame_start in 1:10:50
+]
+```
+
+I've basically added four more blobs by using `for frame_start in 1:10:50` inside the array definition.
+Then I defined the start of the `Action` with: `frame_start:frame_start+149` such that every blob is there for 150 frames and reduced the number of frames in the subactions a bit to have 150 frames.
```
Comment (TheCedarPrince) [inline docs/src/tutorials/tutorial_6.md:171]: At first, this syntax was not immediately apparent as a list comprehension. Really cool thing here! Maybe explicitly state this is a list comprehension being generated by the for loop?

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Pretty awesome tutorial! Just some options to consider and some ideas. Few typos too. Great work @Wikunia !

File: docs/src/tutorials/tutorial_6.md (hunk 2 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:10]: I don't wanna show the final gif at the beginning as I think it destroys the joy a bit :joy: Do you think that's okay?

File: docs/src/tutorials/tutorial_6.md [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+
+## How about Color?
+
+Okay everything is dull when we only use white and black. Let's make it such that our blobs change color from red over cyan to black which also make them disappear.
+
+```julia
+color_anim = Animation(
+    [0, 0.5, 1], # must go from 0 to 1
+    [Lab(colorant"red"), Lab(colorant"cyan"), Lab(colorant"black")],
+    [sineio(), sineio()],
```
[the remark this answers is not in the document]
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:183]: Isn't this explained in: "And we need two easing functions: One defines the movement from red to cyan and the second from cyan to black." ?

File: docs/src/tutorials/tutorial_6.md (hunk 1 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md]: Good catch. Should be fixed

File: docs/src/tutorials/tutorial_6.md (hunk 3 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md]: Should be fixed now. I rephrased it a bit.

File: docs/src/tutorials/tutorial_6.md (hunk 5 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:56]: I explained more and have a plot now. Good point!

File: docs/src/tutorials/tutorial_6.md (hunk 6 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:60]: I've added how to install them which I hope is fine or do you want a formal introduction in what they do? I think Animations.jl is explained in the overall tutorial and Colors.jl is quite clear by the name I hope :smile: At least the functionality I use.

File: docs/src/tutorials/tutorial_6.md (hunk 8 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:152]: I've added a plot now!

File: docs/src/tutorials/tutorial_6.md (hunk 9 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md]: Added a small sentence! I use this quite often in my animations now as it makes everything quite simple

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Just some general thoughts - overall, just about perfect! Few things left to discuss/tweak.

File: docs/src/tutorials/tutorial_6.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:224]: I don't understand what you mean tbh.

[the remark this answers is not in the document]
Comment (TheCedarPrince) [reply docs/src/tutorials/tutorial_6.md:224]: Actually, now looking at this again, I agree with your previous statement about defining movement. Resolved.

File: docs/src/tutorials/tutorial_6.md (hunk 2 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:10]: Do you show what you create though in your tutorials? Or did I show it before in mine :stuck_out_tongue: Why not stick to what we have? Scrolling through it pretty easy/fast. Or am I wrong?

File: docs/src/tutorials/tutorial_6.md (hunk 7 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:98]: Is this resolved?

File: docs/src/tutorials/tutorial_6.md (hunk 6 above)
Comment (Wikunia) [reply docs/src/tutorials/tutorial_6.md:60]: I have added a line for both of them now right under my `] add Animations, Colors` line

File: docs/src/tutorials/tutorial_6.md (hunk 2 above)
Comment (TheCedarPrince) [reply docs/src/tutorials/tutorial_6.md:10]: Hmmm... Actually, we haven't been consistent with this. Instead, I think we should be fine with how it is. Especially given that we do state learning outcomes at the beginning.

**Review by TheCedarPrince — APPROVED** [review]
